common/ast/src/parser/rule/statement.rs

```diff
@@ -47,3 +49,210 @@ pub fn statement(i: Input) -> IResult<Statement> {
+                .into_iter()
+                .reduce(|left, right| {
+                    TableReference::Join(Join {
+                        op: JoinOperator::CrossJoin,
+                        condition: JoinCondition::None,
+                        left: Box::new(left),
+                        right: Box::new(right),
+                    })
+                })
+                .unwrap();
+
+            Query {
+                body: SetExpr::Select(Box::new(SelectStmt {
+                    distinct: distinct.is_some(),
+                    select_list,
+                    from,
+                    selection: where_block.map(|(_, selection)| selection),
+                    group_by: group_by_block
+                        .map(|(_, _, group_by)| group_by)
+                        .unwrap_or_default(),
+                    having: having_block.map(|(_, having)| having),
+                })),
+                order_by: order_by_block
+                    .map(|(_, _, order_by)| order_by)
+                    .unwrap_or_default(),
+                limit: limit_block.map(|(_, limit)| limit).unwrap_or_default(),
+            }
+        },
+    )(i)
+}
+
+pub fn select_target(i: Input) -> IResult<SelectTarget> {
+    let qualified_wildcard = map(
+        rule! {
+            ( #ident ~ "." ~ ( #ident ~ "." )? )? ~ "*"
+        },
+        |(res, _)| match res {
+            Some((fst, _, Some((snd, _)))) => SelectTarget::Indirections(vec![
+                Indirection::Identifier(fst),
+                Indirection::Identifier(snd),
+                Indirection::Star,
+            ]),
+            Some((fst, _, None)) => {
+                SelectTarget::Indirections(vec![Indirection::Identifier(fst), Indirection::Star])
+            }
+            None => SelectTarget::Indirections(vec![Indirection::Star]),
+        },
+    );
+    let projection = map(
+        rule! {
+            #expr ~ ( AS ~ #ident )?
+        },
+        |(expr, alias)| SelectTarget::Projection {
+            expr,
+            alias: alias.map(|(_, name)| name),
+        },
+    );
+
+    rule!(
+        #qualified_wildcard
+        | #projection
+    )(i)
+}
+
+pub fn aliased_table(i: Input) -> IResult<TableReference> {
+    let table_alias = map(rule! { #ident }, |name| TableAlias {
+        name,
+        columns: vec![],
+    });
+
+    map(
+        rule! {
+            #ident ~ ( "." ~ #ident )? ~ ( AS? ~ #table_alias )?
+        },
+        |(fst, snd, alias)| {
+            let (database, table) = match (fst, snd) {
+                (database, Some((_, table))) => (Some(database), table),
+                (table, None) => (None, table),
+            };
+
+            TableReference::Table {
+                database,
+                table,
+                alias: alias.map(|(_, alias)| alias),
+            }
+        },
+    )(i)
+}
+
+pub fn table_reference(i: Input) -> IResult<TableReference> {
+    rule! (
+        #parenthesized_joined_tables
```

Comment:
A `table_reference` could also be `parenthesized_subquery`.

Reply:
addressed